### ISSUE TITLE
EngineCrash [1767] at [rts/System/LoadSave/DemoRecorder.cpp:109]

### ISSUE THREAD

**Issue by Beherith** [beyond-all-reason/spring#836]
**URL of infolog.txt**
https://server4.beyondallreason.info/teiserver/reports/infolog/40087
**Engine Version**
105.1.1-1767-gaaf2cc3105

**Translated Stack Trace with links**

```
<html><body>
<!--StartFragment--><h1>translated with links to github source ('Spring 105.1.1-1767-gaaf2cc3 BAR105 ' detected)</h1>


module | address | file
-- | -- | --
f2cc3 bar\spring.exe | 0x00007ff62fdf4817 | rts/System/LoadSave/DemoRecorder.cpp :109 (https://github.com/beyond-all-reason/spring/blob/aaf2cc3/rts/System/LoadSave/DemoRecorder.cpp#L109)
f2cc3 bar\spring.exe | 0x00007ff62f853348 | rts/Game/PreGame.cpp :604 (https://github.com/beyond-all-reason/spring/blob/aaf2cc3/rts/Game/PreGame.cpp#L604)
f2cc3 bar\spring.exe | 0x00007ff62f8540d1 | /mxe/usr/lib/gcc/x86_64-w64-mingw32.static/11.2.0/include/c++/bits/shared_ptr_base.h :704
f2cc3 bar\spring.exe | 0x00007ff62f854b2d | rts/Game/PreGame.cpp :205 (https://github.com/beyond-all-reason/spring/blob/aaf2cc3/rts/Game/PreGame.cpp#L205)
f2cc3 bar\spring.exe | 0x00007ff62fe1d0b6 | rts/System/SpringApp.cpp :843 (https://github.com/beyond-all-reason/spring/blob/aaf2cc3/rts/System/SpringApp.cpp#L843)
f2cc3 bar\spring.exe | 0x00007ff62fe207c8 | rts/System/SpringApp.cpp :884 (https://github.com/beyond-all-reason/spring/blob/aaf2cc3/rts/System/SpringApp.cpp#L884)
f2cc3 bar\spring.exe | 0x00007ff62fe00f2c | rts/System/Main.cpp :48 (https://github.com/beyond-all-reason/spring/blob/aaf2cc3/rts/System/Main.cpp#L48)
f2cc3 bar\spring.exe | 0x00007ff62f7e13c1 | /mxe/tmp-gcc-x86_64-w64-mingw32.static/gcc-11.2.0.build_/mingw-w64-v9.0.0/mingw-w64-crt/crt/crtexe.c :323
f2cc3 bar\spring.exe | 0x00007ff62f7e14d6 | /mxe/tmp-gcc-x86_64-w64-mingw32.static/gcc-11.2.0.build_/mingw-w64-v9.0.0/mingw-w64-crt/crt/crtexe.c :178
s\SYSTEM32\ntdll.dll | 0x00000000000526a1 | ?? :0

<!--EndFragment-->
</body>
</html>

```

**Additional context**
I see very many crashes of this nature in the reports recently. 

Happens generally on loading:

```
[t=00:01:00.950869][f=-000001] 	file="bitmaps/projectiletextures/barflame02.tga" indx=5895 inflSize=24576kb readTime=98ms
[t=00:01:00.955234][f=-000001] Error: Spring 105.1.1-1767-gaaf2cc3 BAR105 has crashed.
[t=00:01:01.036370][f=-000001] Error: DLL information:
[t=00:01:01.036597][f=-000001] Error: 0x00007ff6e9bf0000	spring
```
```
[t=00:28:42.736162][f=-000001] 	file="luaui/images/decals_gl4/groundscars/t_groundcrack_09_n.tga" indx=7984 inflSize=16384kb readTime=67ms
[t=00:28:42.736175][f=-000001] 	file="luaui/images/decals_gl4/groundscars/t_groundcrack_21_n.tga" indx=8054 inflSize=16384kb readTime=66ms
[t=00:28:42.740037][f=-000001] Error: Spring 105.1.1-1767-gaaf2cc3 BAR105 has crashed.
```

**Comment by Beherith** [comment]
Sometimes its one line above, like:

https://server4.beyondallreason.info/teiserver/reports/infolog/40068

```
\System32\msvcrt.dll 0x000000000007974e ??:0
f2cc3 bar\spring.exe 0x00007ff66af6bf60 strsafe.c:1
f2cc3 bar\spring.exe 0x00007ff66af6bd6e strsafe.c:1
f2cc3 bar\spring.exe 0x00007ff66a243348 rts/Game/PreGame.cpp:604
f2cc3 bar\spring.exe 0x00007ff66a2440d1 /mxe/usr/lib/gcc/x86_64-w64-mingw32.static/11.2.0/include/c++/bits/shared_ptr_base.h:704
f2cc3 bar\spring.exe 0x00007ff66a244b2d rts/Game/PreGame.cpp:205
f2cc3 bar\spring.exe 0x00007ff66a80d0b6 rts/System/SpringApp.cpp:843
f2cc3 bar\spring.exe 0x00007ff66a8107c8 rts/System/SpringApp.cpp:884
f2cc3 bar\spring.exe 0x00007ff66a7f0f2c rts/System/Main.cpp:48
f2cc3 bar\spring.exe 0x00007ff66a1d13c1 /mxe/tmp-gcc-x86_64-w64-mingw32.static/gcc-11.2.0.build_/mingw-w64-v9.0.0/mingw-w64-crt/crt/crtexe.c:323
f2cc3 bar\spring.exe 0x00007ff66a1d14d6 /mxe/tmp-gcc-x86_64-w64-mingw32.static/gcc-11.2.0.build_/mingw-w64-v9.0.0/mingw-w64-crt/crt/crtexe.c:178
S\SYSTEM32\ntdll.dll 0x000000000005a9f8 ??:0

```

**Comment by lhog** [comment]
I noticed this one too. https://github.com/beyond-all-reason/spring/commit/92a5b9dcaa3006a771e0905fab3cc10b8ca126f8 should help seeing what's going on.

**Comment by sprunk** [comment]
Assuming fixed by 92a5b9dcaa3006a771e0905fab3cc10b8ca126f8
* added `length > 0` check which prevents an empty string (or one entirely composed of `\0`s) from iterating into negatives
* apparently no reports since
